karapace/protobuf/io.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+# -*- coding: utf-8 -*-
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import BytesIO
+from karapace.protobuf.exception import IllegalArgumentException, ProtobufSchemaResolutionException, ProtobufTypeException
+from karapace.protobuf.message_element import MessageElement
+from karapace.protobuf.protobuf_to_dict import dict_to_protobuf, protobuf_to_dict
+from karapace.protobuf.schema import ProtobufSchema
+from karapace.protobuf.type_element import TypeElement
+
+import hashlib
+import importlib
+import importlib.util
+import logging
+import os
+
+ZERO_BYTE = b'\x00'
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_class_name(name: str) -> str:
+    return "c_" + hashlib.md5(name.encode('utf-8')).hexdigest()
+
+
+class ProtobufDatumReader:
+    """Deserialize Protobuf-encoded data into a Python data structure."""
+
+    @staticmethod
+    def check_props(schema_one, schema_two, prop_list):
+        for prop in prop_list:
+            if getattr(schema_one, prop) != getattr(schema_two, prop):
+                return False
+        return True
+
+    @staticmethod
+    def match_schemas(writer_schema: ProtobufSchema, reader_schema: ProtobufSchema) -> bool:
+        # TODO (serge): schema comparison by fields required
+
+        if str(writer_schema) == str(reader_schema):
```

Comment:
seems like this one can be simplified to `return str(writer_schema) == str(reader_schema)`

Reply:
yes